cmake/tpl/FindSUPERLUDIST.cmake

```diff
@@ -14,94 +14,103 @@
+      file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _index_size_32 REGEX "#undef XSDK_INDEX_SIZE")
+      if(_index_size_64)
+        set(SUPERLUDIST_INDEX_SIZE 64 CACHE STRING "SuperLU DIST index size (bit width)" FORCE)
+      else()
+        set(SUPERLUDIST_INDEX_SIZE 32 CACHE STRING "SuperLU DIST index size (bit width)" FORCE)
+      endif()
+      mark_as_advanced(FORCE SUPERLUDIST_INDEX_SIZE)
+  else()
+    file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_with_index_size REGEX "XSDK_INDEX_SIZE")
+    list(GET _strings_with_index_size 0 _index_size_string)
+    string(REGEX MATCHALL "[0-9][0-9]" SUPERLUDIST_INDEX_SIZE "${_index_size_string}")
+  endif()
+  file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_have_cuda REGEX "HAVE_CUDA")
+  string(REGEX MATCH "TRUE|FALSE" _has_cuda "${_strings_have_cuda}")
+  file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_have_rocm REGEX "HAVE_ROCM")
+  string(REGEX MATCH "TRUE|FALSE" _has_rocm "${_strings_have_rocm}")
+  set(SUPERLUDIST_CUDA ${_has_cuda} CACHE INTERNAL "SuperLU DIST was built with CUDA support" FORCE)
+  set(SUPERLUDIST_ROCM ${_has_rocm} CACHE INTERNAL "SuperLU DIST was built with ROCm support" FORCE)
+  unset(_has_cuda)
+  unset(_has_rocm)
 endif()
 
 # find the library version file
-if(SUPERLUDIST_LIBRARY AND SUPERLUDIST_INCLUDE_DIR)
-  find_file(SUPERLUDIST_VERSION_PATH superlu_defs.h PATHS ${SUPERLUDIST_INCLUDE_DIR})
+if(NOT SUPERLUDIST_VERSION AND SUPERLUDIST_INCLUDE_DIRS)
+  find_file(SUPERLUDIST_VERSION_PATH superlu_defs.h PATHS "${SUPERLUDIST_INCLUDE_DIRS}")
 
-  file(STRINGS ${SUPERLUDIST_VERSION_PATH} _version_major REGEX "SUPERLU_DIST_MAJOR_VERSION")
+  file(STRINGS "${SUPERLUDIST_VERSION_PATH}" _version_major REGEX "SUPERLU_DIST_MAJOR_VERSION")
   list(GET _version_major 0 _version_string)
   string(REGEX MATCHALL "[0-9]" _version_major "${_version_string}")
 
-  file(STRINGS ${SUPERLUDIST_VERSION_PATH} _version_minor REGEX "SUPERLU_DIST_MINOR_VERSION")
+  file(STRINGS "${SUPERLUDIST_VERSION_PATH}" _version_minor REGEX "SUPERLU_DIST_MINOR_VERSION")
   list(GET _version_minor 0 _version_string)
   string(REGEX MATCHALL "[0-9]" _version_minor "${_version_string}")
 
-  file(STRINGS ${SUPERLUDIST_VERSION_PATH} _version_patch REGEX "SUPERLU_DIST_PATCH_VERSION")
+  file(STRINGS "${SUPERLUDIST_VERSION_PATH}" _version_patch REGEX "SUPERLU_DIST_PATCH_VERSION")
   list(GET _version_patch 0 _version_string)
   string(REGEX MATCHALL "[0-9]" _version_patch "${_version_string}")
 
   set(SUPERLUDIST_VERSION "${_version_major}.${_version_minor}.${_version_patch}")
   mark_as_advanced(FORCE SUPERLUDIST_VERSION_PATH)
 endif()
 
+# add libraries for OpenMP support
+if(SUPERLUDIST_OpenMP)
+  list(APPEND SUPERLUDIST_LINK_LIBRARIES OpenMP::OpenMP_CXX)
+endif()
+# add libraries for GPU support
+if(SUPERLUDIST_CUDA)
+  list(APPEND SUPERLUDIST_LINK_LIBRARIES CUDA::cudart CUDA::cublas)
+endif()
+if(SUPERLUDIST_HIP)
+  list(APPEND SUPERLUDIST_LINK_LIBRARIES roc::hipblas)
+endif()
```

Comment:
Need to make sure OpenMP, CUDA, and/or HIP is enabled/found so these targets exists.